### ISSUE TITLE
Impeller produce black screen in android real device

### ISSUE THREAD

**Issue by sfad** [flutter/flutter#160647]
### Steps to reproduce

just create new app and run it on Samsung galaxy A51 with flutter 3.27

but when I disable Impeller it work fine.

In emulator it work fine also.
in iOS work fine without any issue.

The test device is Galaxy A51
Android version 13

### Expected results

default counter app

### Actual results

Black screen

### Code sample

<details open><summary>Code sample</summary>

```dart
no code needed issue with the renderer engine.
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console

Launching lib/main.dart on SM A515F in debug mode...
Warning: SDK processing. This version only understands SDK XML versions up to 3 but an SDK XML file of version 4 was encountered. This can happen if you use versions of Android Studio and the command-line tools that were released at different times.
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
3 warnings
✓ Built build/app/outputs/flutter-apk/app-debug.apk
I/flutter (18092): [IMPORTANT:flutter/shell/platform/android/android_context_vk_impeller.cc(60)] Using the Impeller rendering backend (Vulkan).
Connecting to VM Service at ws://127.0.0.1:60418/Wg-2rDOzosA=/ws
Connected to the VM Service.
D/SurfaceView@feadd95(18092):  setAlpha: mUseAlpha = false alpha=1.0
D/SurfaceView@feadd95(18092):  updateSurfaceAlpha: setUseAlpha() is not called, ignored.
I/ViewRootImpl@78e6c6f[MainActivity](18092): Setup new sync id=0
I/ViewRootImpl@78e6c6f[MainActivity](18092): Setting syncFrameCallback
I/ViewRootImpl@78e6c6f[MainActivity](18092): registerCallbacksForSync syncBuffer=false
D/SurfaceView@feadd95(18092): updateSurfacePosition RenderWorker, frameNr = 1, position = [0, 0, 1080, 2274] surfaceSize = 1080x2274
I/SurfaceView@feadd95(18092): uSP: rtp = Rect(0, 0 - 1080, 2274) rtsw = 1080 rtsh = 2274
I/SurfaceView@feadd95(18092): onSSPAndSRT: pl = 0 pt = 0 sx = 1.0 sy = 1.0
I/ViewRootImpl@78e6c6f[MainActivity](18092): mWNT: t=0x7e74eb0400 mBlastBufferQueue=0x7de87cc100 fn= 1 caller= android.view.ViewRootImpl.lambda$applyTransactionOnDraw$11$android-view-ViewRootImpl:12865 android.view.ViewRootImpl$$ExternalSyntheticLambda1.onFrameDraw:4 android.graphics.HardwareRenderer$FrameDrawingCallback.onFrameDraw:924
I/ViewRootImpl@78e6c6f[MainActivity](18092): Received frameDrawingCallback syncResult=0 frameNum=1.
I/ViewRootImpl@78e6c6f[MainActivity](18092): Setting up sync and frameCommitCallback
I/SurfaceView@feadd95(18092): aOrMT: ViewRootImpl@78e6c6f[MainActivity] t = android.view.SurfaceControl$Transaction@ed7e19d fN = 1 android.view.SurfaceView.-$$Nest$mapplyOrMergeTransaction:0 android.view.SurfaceView$SurfaceViewPositionUpdateListener.positionChanged:1537 android.graphics.RenderNode$CompositePositionUpdateListener.positionChanged:373
I/ViewRootImpl@78e6c6f[MainActivity](18092): mWNT: t=0x7e74eb0300 mBlastBufferQueue=0x7de87cc100 fn= 1 caller= android.view.SurfaceView.applyOrMergeTransaction:1455 android.view.SurfaceView.-$$Nest$mapplyOrMergeTransaction:0 android.view.SurfaceView$SurfaceViewPositionUpdateListener.positionChanged:1537
I/BLASTBufferQueue(18092): [ViewRootImpl@78e6c6f[MainActivity]#0](f:0,a:0) onFrameAvailable the first frame is available
I/ViewRootImpl@78e6c6f[MainActivity](18092): Received frameCommittedCallback lastAttemptedDrawFrameNum=1 didProduceBuffer=true
D/OpenGLRenderer(18092): CFMS:: SetUp Pid : 18092    Tid : 18125
I/ViewRootImpl@78e6c6f[MainActivity](18092): onSyncComplete
I/ViewRootImpl@78e6c6f[MainActivity](18092): setupSync seqId=0 mSyncId=0 fn=1 caller=android.view.ViewRootImpl$$ExternalSyntheticLambda11.accept:6 android.window.SurfaceSyncer.lambda$setupSync$1$android-window-SurfaceSyncer:128 android.window.SurfaceSyncer$$ExternalSyntheticLambda1.accept:8 android.window.SurfaceSyncer$SyncSet.checkIfSyncIsComplete:382 android.window.SurfaceSyncer$SyncSet.markSyncReady:359 android.window.SurfaceSyncer.markSyncReady:151 android.view.ViewRootImpl.performTraversals:4503
I/ViewRootImpl@78e6c6f[MainActivity](18092): reportDrawFinished seqId=0 mSyncId=-1 fn=1 mSurfaceChangedTransaction=0x7de838ef00
I/DecorView(18092): notifyKeepScreenOnChanged: keepScreenOn=false
I/ViewRootImpl@78e6c6f[MainActivity](18092): performTraversals params={(0,0)(fillxfill) sim={adjust=resize forwardNavigation} ty=BASE_APPLICATION fmt=TRANSLUCENT wanim=0x1030001
I/ViewRootImpl@78e6c6f[MainActivity](18092):   fl=81810100
I/ViewRootImpl@78e6c6f[MainActivity](18092):   pfl=16020040
I/ViewRootImpl@78e6c6f[MainActivity](18092):   vsysui=2500
I/ViewRootImpl@78e6c6f[MainActivity](18092):   apr=LIGHT_STATUS_BARS
I/ViewRootImpl@78e6c6f[MainActivity](18092):   bhv=DEFAULT
I/ViewRootImpl@78e6c6f[MainActivity](18092):   fitSides= naviIconColor=0}
I/ViewRootImpl@78e6c6f[MainActivity](18092): performTraversals mFirst=false windowShouldResize=false viewVisibilityChanged=false mForceNextWindowRelayout=false params={(0,0)(fillxfill) sim={adjust=resize forwardNavigation} ty=BASE_APPLICATION fmt=TRANSLUCENT wanim=0x1030001
I/ViewRootImpl@78e6c6f[MainActivity](18092):   fl=81810100
I/ViewRootImpl@78e6c6f[MainActivity](18092):   pfl=16020040
I/ViewRootImpl@78e6c6f[MainActivity](18092):   vsysui=2500
I/ViewRootImpl@78e6c6f[MainActivity](18092):   apr=LIGHT_STATUS_BARS
I/ViewRootImpl@78e6c6f[MainActivity](18092):   bhv=DEFAULT
I/ViewRootImpl@78e6c6f[MainActivity](18092):   fitSides= naviIconColor=0}
I/ViewRootImpl@78e6c6f[MainActivity](18092): updateBlastSurfaceIfNeeded mBlastBufferQueue=0x7de87cc100 isSameSurfaceControl=true
I/BLASTBufferQueue(18092): update, w= 1080 h= 2400 mName = ViewRootImpl@78e6c6f[MainActivity] mNativeObject= 0x7de87cc100 sc.mNativeObject= 0x7e6ffac5c0 format= -3 caller= android.view.ViewRootImpl.updateBlastSurfaceIfNeeded:2898 android.view.ViewRootImpl.relayoutWindow:9847 android.view.ViewRootImpl.performTraversals:3884 android.view.ViewRootImpl.doTraversal:3116 android.view.ViewRootImpl$TraversalRunnable.run:10885 android.view.Choreographer$CallbackRecord.run:1301
I/ViewRootImpl@78e6c6f[MainActivity](18092): Relayout returned: old=(0,0,1080,2400) new=(0,0,1080,2400) req=(1080,2400)0 dur=15 res=0x0 s={true 0x7de2708800} ch=false seqId=0
I/ViewRootImpl@78e6c6f[MainActivity](18092): updateBoundsLayer: t = android.view.SurfaceControl$Transaction@5171612 sc = Surface(name=Bounds for - com.neatit.vendors_manage/com.neatit.vendors_manage.MainActivity@0)/@0x77a11e3 frame = 2
I/ViewRootImpl@78e6c6f[MainActivity](18092): mWNT: t=0x7de84b6000 mBlastBufferQueue=0x7de87cc100 fn= 2 caller= android.view.ViewRootImpl.prepareSurfaces:2985 android.view.ViewRootImpl.performTraversals:4233 android.view.ViewRootImpl.doTraversal:3116
I/ViewRootImpl@78e6c6f[MainActivity](18092): MSG_WINDOW_FOCUS_CHANGED 1 0
I/ViewRootImpl@78e6c6f[MainActivity](18092): mThreadedRenderer.initializeIfNeeded()#2 mSurface={isValid=true 0x7de2708800}
D/InputMethodManager(18092): startInputInner - Id : 0
I/InputMethodManager(18092): startInputInner - mService.startInputOrWindowGainedFocus
D/InputMethodManager(18092): startInputInner - Id : 0
D/InsetsController(18092): onStateChanged: InsetsState: {mDisplayFrame=Rect(0, 0 - 1080, 2400), mDisplayCutout=DisplayCutout{insets=Rect(0, 88 - 0, 0) waterfall=Insets{left=0, top=0, right=0, bottom=0} boundingRect={Bounds=[Rect(0, 0 - 0, 0), Rect(512, 0 - 568, 88), Rect(0, 0 - 0, 0), Rect(0, 0 - 0, 0)]} cutoutPathParserInfo={CutoutPathParserInfo{displayWidth=1080 displayHeight=2400 physicalDisplayWidth=1080 physicalDisplayHeight=2400 density={2.625} cutoutSpec={M 0,0 M 0,12.25101276465640 a 10.60613009248640,10.60613009248640 0 1,0 0,21.21226018497290 a 10.60613009248640,10.60613009248640 0 1,0 0,-21.21226018497290 Z @dp} rotation={0} scale={1.0} physicalPixelDisplaySizeRatio={1.0}}}}, mRoundedCorners=RoundedCorners{[RoundedCorner{position=TopLeft, radius=84, center=Point(84, 84)}, RoundedCorner{position=TopRight, radius=84, center=Point(996, 84)}, RoundedCorner{position=BottomRight, radius=84, center=Point(996, 2316)}, RoundedCorner{position=BottomLeft, radius=84, center=Point(84, 2316)}]}  mRoundedCornerFrame=Rect(0, 0 - 1080, 2400), mPrivacyIndicatorBounds=PrivacyIndicatorBounds {static bounds=Rect(964, 0 - 1080, 88) rotation=0}, mSources= { InsetsSource: {mType=ITYPE_STATUS_BAR, mFrame=[0,0][1080,88], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_NAVIGATION_BAR, mFrame=[0,2274][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_LEFT_GESTURES, mFrame=[0,0][0,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_RIGHT_GESTURES, mFrame=[1080,0][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_TOP_MANDATORY_GESTURES, mFrame=[0,0][1080,120], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_BOTTOM_MANDATORY_GESTURES, mFrame=[0,2274][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_LEFT_DISPLAY_CUTOUT, mFrame=[0,0][-100000,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_TOP_DISPLAY_CUTOUT, mFrame=[0,0][1080,88], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_RIGHT_DISPLAY_CUTOUT, mFrame=[100000,0][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_BOTTOM_DISPLAY_CUTOUT, mFrame=[0,100000][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_TOP_TAPPABLE_ELEMENT, mFrame=[0,0][1080,88], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_BOTTOM_TAPPABLE_ELEMENT, mFrame=[0,2274][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_IME, mFrame=[0,0][0,0], mVisibleFrame=[0,2274][1080,2400], mVisible=false, mInsetsRoundedCornerFrame=false} } host=com.neatit.vendors_manage/com.neatit.vendors_manage.MainActivity from=android.view.ViewRootImpl$ViewRootHandler.handleMessageImpl:6740
D/ProfileInstaller(18092): Installing profile for com.neatit.vendors_manage
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 0
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 1
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 0
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 1
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 0
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 1
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 0
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 1
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 0
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 1
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 0
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 1
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 0
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 1
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 0
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 1
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 0
I/ViewRootImpl@78e6c6f[MainActivity](18092): ViewPostIme pointer 1
Application finished.

Exited.


```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.27.0, on macOS 15.2 24C101 darwin-arm64, locale
    en-BH)
[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 16.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.2)
[✓] VS Code (version 1.96.1)
[✓] Connected device (2 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @sfad, this should be fixed by the stable 3.27.1 hot fix. Can you upgrade to this version to see if it is indeed fixed for you?

**Comment by sfad** [comment]
> Hi @sfad, this should be fixed by the stable 3.27.1 hot fix. Can you upgrade to this version to see if it is indeed fixed for you?

Yes It dose solve the problem in this model but I don't know if it safe to make the impeller the default before checking other devices because there are many in android. 

I love a lot the new features added in recent update.

**Comment by sfad** [comment]
Thanks for your support
